stock_generate_putaway_from_inventory/models/stock_inventory.py

```diff
@@ -0,0 +1,89 @@
+# © 2018 Akretion
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, _
+from odoo.exceptions import Warning as UserError
+
+
+class LocationMissingPutawayExc(Exception):
+    pass
+
+
+class InventoryNotValidatedExc(Exception):
+    pass
+
+
+class StockInventory(models.Model):
+    _inherit = "stock.inventory"
+
+    def _handle_errors(self, missing_putaway, not_validated):
+        msg = ""
+        if missing_putaway:
+            msg += _(
+                "Some locations don't have a putaway strategy. "
+                "Please set a putaway strategy for the locations: "
+            )
+            for el in missing_putaway.mapped("name"):
+                msg += "%s ," % el
+            msg = msg[:-2]
+            msg += "\n\n"
+        if not_validated:
+            msg += _(
+                "Some inventories are not validated. Please "
+                "validate the following inventories: "
+            )
+            for el in not_validated:
+                msg += "%s ," % el.name
+            msg = msg[:-2]
+        return msg
+
+    def action_generate_putaway_strategy(self):
+        missing_putaway = self.env["stock.location"]
+        not_validated = self.env["stock.inventory"]
+        for inventory in self:
+            try:
+                inventory._generate_putaway_strategy()
```

Comment:
I think we should keep this as simple a possible.
In the huge majority of case the user will generate putaway only for one inventory (most of customer create one inventory per year).
I think the code will be a lot simplify if we simplify raise directly the error.
So just do 
```
for inventory in self:
       inventory._generate_putaway_strategy()
```
And directly raise error in the method "_generate_putaway_strategy" then you can drop the handle_errors and so drop 16~20 line of code

Reply:
fixed at https://github.com/akretion/stock-logistics-warehouse/commit/f3159eba294a08566e4c752fe41cb6c6c7b1df9b